### PR TITLE
Add token config

### PR DESCRIPTION
**Pull request by Virgil993**
This PR changes the way authentication is done by using the pulumi config feature to set the token as a secret in the pulumi.yaml file associated with the stack. Will have to change this a little in the feature to use access tokens as opposed to session tokens.

### CODE REVIEW

File: provider/domain/user_payload.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+	ID string `json:"id"`
+	Email string `json:"email"`
+	ProfileUrl string `json:"profileUrl"`
+	SubscriptionPlan string `json:"subscriptionPlan"`
+	SubscriptionPrice string `json:"subscriptionPrice"`
+	MemberSince string `json:"memberSince"`
+	SubscriptionLimits SubscriptionLimits `json:"subscriptionLimits"`
+	CustomSubscription bool `json:"customSubscription"`
+	OnboardingInfo OnboardingInfo `json:"onboardingInfo"`
+}
```
Comment (andreia-oca) [inline provider/domain/user_payload.go:28]: Why is this here? Seems not used to me.

Also we don't want to bring so much complexity from the backend into this pulumi provider

File: provider/requests/get_frontend_presigned_url.go [hunk 2]
```diff
@@ -44,7 +46,7 @@ func GetFrontendPresignedUrl(subdomain string, projectName string, stage string,
 	}
 
 	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("Authorization", "Bearer "+authToken)
+	req.Header.Set("Authorization", "Bearer "+ctx.Value("authToken").(string))
```
Comment (andreia-oca) [inline provider/requests/get_frontend_presigned_url.go:49]: You can use `GetAuthToken` to increase visibility and also check if the `authToken` != "".

I would also do something like:
```
token := GetAuthToken(ctx);
...
req.Header.Set("Authorization", "Bearer "+token)
```

File: provider/resources/database.go [hunk 3]
```diff
@@ -26,14 +26,19 @@ type DatabaseState struct {
 }
 
 func (*Database) Read(ctx p.Context, id string, inputs DatabaseArgs, state DatabaseState) (string, DatabaseArgs, DatabaseState , error) {
+	authToken, err := utils.IsLoggedIn(ctx)
+	if err != nil {
+		return id, inputs, state, err
+	}
+	ctx = p.CtxWithValue(ctx, "authToken", authToken)
```
Comment (andreia-oca) [inline provider/resources/database.go:33]: It seems to me that `AuthToken` is already in the p.Context.

I assume that pulumi populates the context because I see that this line of code is possible:
```
infer.GetConfig[*domain.Config](ctx).AuthToken
```

It's not necessary for us to add manually the `authToken`. So please remove this line of code and use `GetAuthToken` where possible.

File: provider/resources/project.go [hunk 4]
```diff
@@ -25,13 +25,20 @@ type ProjectState struct {
 }
 
 func (*Project) Create(ctx p.Context, name string, input ProjectArgs, preview bool) (string, ProjectState, error) {
+	authToken, err := utils.IsLoggedIn(ctx)
+	if err != nil {
+		return name, ProjectState{}, err
+	}
```
Comment (andreia-oca) [inline provider/resources/project.go:31]: This is not great because we have to add it to every method. This should be implemented like some sort of middleware that will automatically check the token before calling `Create`.

I investigated a bit and I am not sure how to inject a middleware between the provider and each resource.

But I prefer to leave out this check rather than having it for every method.

**Review by andreia-oca — APPROVED** [review]

File: provider/requests/get_frontend_presigned_url.go (hunk 2 above)
Comment (Virgil993) [reply provider/requests/get_frontend_presigned_url.go:49]: We check at the begining of every opperation if the user is logged in. That also retrieves the token that's why i use it directly from context and why i don't think it's neccesary to check if the authToken != "" . That being said, if it's just for readability purposes i can change it no problems.

File: provider/resources/database.go (hunk 3 above)
Comment (Virgil993) [reply provider/resources/database.go:33]: That's true I didn't notice it, i will make the change

File: provider/resources/project.go (hunk 4 above)
Comment (Virgil993) [reply provider/resources/project.go:31]: I agree that this should be a middleware but i haven't found anything that can act as a middleware for pulumi resources, will investigate further. So for now we just rely on the backend to return an unauthorized status code if the token is not valid right?

File: provider/domain/user_payload.go (hunk 1 above)
Comment (Virgil993) [reply provider/domain/user_payload.go:28]: I tried to keep the logic from the CLI as we might at some point need the user response from the GetUser request. I will remove it so we don't get the codebase bloated
